### PR TITLE
rds snapshot class

### PR DESCRIPTION
**Pull request by joshuaroot**
added class for rds instance snapshots

added class for rds retention window (enable automated rds backups)

### CODE REVIEW

File: c7n/resources/rds.py [hunk 1]
```diff
@@ -175,5 +176,59 @@ def process(self, resources):
...
+            if v == 0:
+                    self.retention_window(resource)
+        except ValueError:
+                self.set_retention_window(resource)
+
+    def set_retention_window(self, resource):
+        c = local_session(self.manager.session_factory).client('rds')
+        c.modify_db_instance(
+            DBInstanceIdentifier=resource['DBInstanceIdentifier'],
+            BackupRetentionPeriod=7)
```
Comment (kapilt) [inline c7n/resources/rds.py]: this should be configurable source, from self.data['days']

also these need schema attributes to be valid from config

**Comment by kapilt** [comment]
looks good, looks it missed the schema work for the validation to take place though.

File: c7n/resources/rds.py [hunk 2]
```diff
@@ -223,5 +224,66 @@ def process(self, resources):
...
+
+    date_attribute = "BackupRetentionPeriod"
+    schema = type_schema('retention', days={'type': 'number'})
+
+    def process(self, resources):
+        with self.executor_factory(max_workers=3) as w:
+            futures = []
+            for resource in resources:
+                futures.append(w.submit(
+                    self.process_rds_snapshot,
```
Comment (kapilt) [inline c7n/resources/rds.py]: wrong method name to submit.

its really helpful if you at least try to test things before submitting pull request.

File: c7n/resources/rds.py [hunk 3]
```diff
@@ -223,5 +224,66 @@ def process(self, resources):
...
+                    resource))
+                for f in as_completed(futures):
+                    if f.exception():
+                        self.log.error(
+                            "Exception setting rds retention  \n %s" % (
+                                f.exception()))
+
+    def process_snapshot_retention(self, resource):
+        try:
+            v = int(resource['BackupRetentionPeriod'])
```
Comment (kapilt) [inline c7n/resources/rds.py]: instead of try except here you can do v = int(resource.get('BackupRetentionPeriod', 0))

**Comment by joshuaroot** [comment]
made adjustments as directed. tested OK in sandbox

**Pull request merged**